workshops/nextjs_dashboard/README.md

```diff
@@ -0,0 +1,344 @@
+    }
+
+    navigator.geolocation.getCurrentPosition(success, error)
+  }
+
+  render() {
+    const { currently, forecast } = this.state
+    return (
+      <div>
+        <h2>Weather</h2>
+        {currently === 'loading' ? (
+          <p>Loading…</p>
+        ) : currently === 'error' ? (
+          <p>There was an error :(</p>
+        ) : (
+          <p>The current temperature is: {forecast.currently.temperature}º</p>
+        )}
+      </div>
+    )
+  }
+}
+
+export default Weather
+```
+
+On the `const url` line, you'll notice we're using this other URL with `cors-anywhere` in it. This is an inelegant hack to get our site working: Dark Sky only wants you to make "proper" requests from your servers, but since we just want to get this working in our browser, we're using this other utility.
+
+## Getting the news headlines
+
+We'll be using the News API now, with a very similar structure to the Weather component. We need to fetch the news, save it to state, then render the list of articles. Open `components/News.js`:
+
+```js
+import React, { Component } from 'react'
+import axios from 'axios'
+
+const API_KEY = '…'
+
+class News extends Component {
+  state = {
+    currently: 'loading',
+    news: []
+  }
+
+  componentDidMount() {
+    const url =
+      'https://newsapi.org/v1/articles?source=google-news&sortBy=top&apiKey='
+
+    axios
+      .get(url + API_KEY)
+      .then(res => res.data)
+      .then(news => this.setState({ news, currently: 'success' }))
+      .catch(() => this.setState({ currently: 'error' }))
+  }
+
+  render() {
+    const { currently, news } = this.state
+    return (
+      <div>
+        <h2>News</h2>
+        {currently === 'loading' ? (
+          <p>Loading…</p>
+        ) : currently === 'error' ? (
+          <p>There was an error :(</p>
+        ) : (
+          <ul>
+            {news.articles.map(article => (
+              <li key={article.url}>
+                <a href={article.url}>{article.title}</a>
+              </li>
+            ))}
+          </ul>
+        )}
+      </div>
+    )
+  }
+}
+
+export default News
+```
+
+You'll need to use the component though :) So head back to `pages/index.js`, and add `import News from '../components/News'` near the top and add the component to the body. Like this:
+
+```js
+import React from 'react'
+import Weather from '../components/Weather'
+import News from '../components/News'
+
+export default () => (
+  <div>
+    <h1>Welcome, YOURNAME!</h1>
+    <Weather />
+    <News />
+    <footer>
+      Powered by <a href="https://newsapi.org/">NewsAPI</a> and{' '}
+      <a href="https://darksky.net/poweredby/">Dark Sky</a>
+    </footer>
+  </div>
+)
+```
+
+## Styling the news
+
+We've got the basics of our news headlines. Let's make it look incredible. Try experimenting with the CSS now! Add a tag of `<style jsx>{\`your css\`}</style>` and play around. Don’t just copy this example, try your own :)
```

Comment:
Looks like this markdown doesn't do what you want it to. Copy, pasted:

`<style jsx>{\`your css\`}</style>`